aw_transform/flood.py

```diff
@@ -15,38 +15,42 @@ def flood(events: List[Event], pulsetime: float=5) -> List[Event]:
 
-        # Sanity check in case events overlap
-        if gap < timedelta(0):
-            if e1.data == e2.data:
-                # Events with negative gap but same data safely be merged
-                start = min(e1.timestamp, e2.timestamp)
-                end = max(e1.timestamp + e1.duration, e2.timestamp + e2.duration)
-                e1.timestamp = start
-                e1.duration = (end - start)
-                e2.timestamp = end
-                e2.duration = timedelta(0)
-                if not warned_about_negative_gap_safe:
-                    logger.warning("Gap was of negative duration ({}s), but could be safely merged. This error will only show once per batch.".format(gap.total_seconds()))
-                    warned_about_negative_gap_safe = True
-                gap = e2.timestamp - (e1.timestamp + e1.duration)
-            elif not warned_about_negative_gap_unsafe:
-                # Events with negative gap but differing data cannot be merged
-                logger.warning("Gap was of negative duration ({}s), and could NOT be safely merged. This error will only show once per batch.".format(gap.total_seconds()))
-                warned_about_negative_gap_unsafe = True
-                # logger.warning("Event 1 (id {}): {} {}".format(e1.id, e1.timestamp, e1.duration))
-                # logger.warning("Event 2 (id {}): {} {}".format(e2.id, e2.timestamp, e2.duration))
+        if not gap:
+            continue
 
-        # if timedelta(0) <= gap <= timedelta(seconds=pulsetime):
-        if timedelta(0) < gap <= timedelta(seconds=pulsetime):
+        # Sanity check in case events overlap
+        if gap < timedelta(0) and e1.data == e2.data:
+            # Events with negative gap but same data can safely be merged
+            start = min(e1.timestamp, e2.timestamp)
+            end = max(e1.timestamp + e1.duration, e2.timestamp + e2.duration)
+            e1.timestamp, e1.duration = start, (end - start)
+            e2.timestamp, e2.duration = end, timedelta(0)
+            if not warned_about_negative_gap_safe:
+                logger.warning("Gap was of negative duration but could be safely merged ({}s). This message will only show once per batch.".format(gap.total_seconds()))
+                warned_about_negative_gap_safe = True
+        elif gap < -negative_gap_trim_thres and not warned_about_negative_gap_unsafe:
+            # Events with negative gap but differing data cannot be merged safely
+            logger.warning("Gap was of negative duration and could NOT be safely merged ({}s). This warning will only show once per batch.".format(gap.total_seconds()))
+            warned_about_negative_gap_unsafe = True
+            # logger.warning("Event 1 (id {}): {} {}".format(e1.id, e1.timestamp, e1.duration))
+            # logger.warning("Event 2 (id {}): {} {}".format(e2.id, e2.timestamp, e2.duration))
+        elif -negative_gap_trim_thres < gap <= timedelta(seconds=pulsetime):
```

Comment:
I don't get it, why is negative_gap_trim_thres needed?
Is timedelta a float?

Reply:
Not really, afaik it represented with an int for seconds and an int for microseconds.

But I noticed there were a few -0.001s gaps appearing in the logs which should be harmless to trim.